ui/src/routes/lecture.tsx

```diff
@@ -24,21 +24,33 @@ function filenameFromUrl(url: string) {
 function Lecture() {
   const [lecture, setLecture] = useState<any>({});
   const [loading, setLoading] = useState(false);
+  const [year, setYear] = useState("");
+  const [indicoId, setIndicoId] = useState("");
   let { lectureId } = useParams();
 
   const fetchLecture = async () => {
     try {
       setLoading(true);
       const results = await getApiRoot().get(`/lectures/${lectureId}/`);
-      setLoading(false);
       setLecture(results.data);
+
+      const year = await results.data.date.slice(0, 4);
+      setYear(year);
+
+      const indicoId = await results.data.event_details.split("/")[4];
+      setIndicoId(indicoId);
+
+      setLoading(false);
+      return;
     } catch (error) {
-      setLecture({});
+      return error;
     }
   };
 
   useEffect(() => {
-    fetchLecture();
+    (async () => {
+      await fetchLecture();
+    })();
```

Comment:
it's not a good practice to have async here

Reply:
Noted.